u8timeseries/timeseries.py

```diff
@@ -271,6 +270,9 @@ def intersect(self, other: 'TimeSeries') -> 'TimeSeries':
         """
         Returns a slice containing the intersection of this TimeSeries and the one provided in argument.
 
+        TODO: This function does not really behave as described. Not really an intersect
+        TODO: In case of no intersection found (or too small), should it raise an error (like now), or return None?
```

Comment:
I would say raise an error

Reply:
Ok, I will leave it like that then.
Another possibility would be to return the common index, instead of the resulting TimeSeries (an actual slice).
Thus, with a __getitem__ method, one could retrieve that slice for any object